From 986087659d42f22616f58e7ab6337f4c4d99fec6 Mon Sep 17 00:00:00 2001
From: Ming Liu-Chan <ming.liu-chan@digi.com>
Date: Thu, 17 Nov 2022 15:43:10 -0500
Subject: [PATCH] ModemManager plugin: updated sierra plugin for vendor string

Similar to DAL-6743, wrong plugin is used when Sierra modem
devices boot up. Added sierra vendor to sierra plugin.

DAL-7045
---
 plugins/sierra/mm-plugin-sierra.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plugins/sierra/mm-plugin-sierra.c b/plugins/sierra/mm-plugin-sierra.c
index e5a61b3c..088930be 100644
--- a/plugins/sierra/mm-plugin-sierra.c
+++ b/plugins/sierra/mm-plugin-sierra.c
@@ -88,6 +88,7 @@ mm_plugin_create (void)
     static const gchar *subsystems[] = { "tty", "net", "usb", NULL };
     static const guint16 vendor_ids[] = { 0x1199, 0 };
     static const gchar *drivers[] = { "qmi_wwan", "cdc_mbim", NULL };
+    static const gchar *vendor_strings[] = { "Sierra Wireless, Incorporated", NULL };
 
     return MM_PLUGIN (
         g_object_new (MM_TYPE_PLUGIN_SIERRA,
@@ -95,6 +96,7 @@ mm_plugin_create (void)
                       MM_PLUGIN_ALLOWED_SUBSYSTEMS, subsystems,
                       MM_PLUGIN_ALLOWED_VENDOR_IDS, vendor_ids,
                       MM_PLUGIN_ALLOWED_DRIVERS,    drivers,
+                      MM_PLUGIN_ALLOWED_VENDOR_STRINGS, vendor_strings,
                       MM_PLUGIN_ALLOWED_AT,         TRUE,
                       MM_PLUGIN_ALLOWED_QMI,        TRUE,
                       MM_PLUGIN_ALLOWED_MBIM,       TRUE,
-- 
2.25.1

